src/irr/asset/CCPUMeshPacker.cpp

```diff
@@ -0,0 +1,63 @@
+#include "irr/asset/CCPUMeshPacker.h"
+#include "irrlicht.h"
+
+namespace irr 
+{ 
+namespace asset 
+{
+
+std::optional<std::pair<ICPUMesh*, DrawElementsIndirectCommand_t>> CCPUMeshPacker::packMeshes(core::vector<ICPUMesh*>& meshes)
+{
+	core::vector<ICPUMeshBuffer*> meshBuffers;
+
+	for (auto* mesh : meshes)
+	{
+		for (uint32_t i = 0; i < mesh->getMeshBufferCount(); i++)
+		{
+			meshBuffers.push_back(mesh->getMeshBuffer(i));
+		}
+	}
+
+	if (meshBuffers.empty())
+	{
+		_IRR_DEBUG_BREAK_IF(true);
+		return {};
+	}
+
+	/*
+	Requirements for input mesh buffers:
```

Comment:
Actually all renderpasses of all the mesh buffers should have identical  settings, need to update validation